tests/pc/test_po_cleanup.py

```diff
@@ -27,36 +27,38 @@ def ignore_expected_loganalyzer_exceptions(rand_one_dut_hostname, loganalyzer):
         ]
         loganalyzer[rand_one_dut_hostname].ignore_regex.extend(ignoreRegex)
         expectRegex = [
-            ".*teamd#teammgrd: :- cleanTeamProcesses.*",
-            ".*teamd#teamsyncd: :- cleanTeamSync.*"
+            ".*teammgrd: :- cleanTeamProcesses.*",
+            ".*teamsyncd: :- cleanTeamSync.*"
         ]
         loganalyzer[rand_one_dut_hostname].expect_regex.extend(expectRegex)
 
-
-def check_kernel_po_interface_cleaned(duthost):
-    res = duthost.shell("ip link show | grep -c PortChannel",  module_ignore_errors=True)["stdout_lines"][0].decode("utf-8")
+def check_kernel_po_interface_cleaned(duthost, asic_index):
+    namespace = duthost.get_namespace_from_asic_id(asic_index)
+    res = duthost.shell(duthost.get_linux_ip_cmd_for_namespace("ip link show | grep -c PortChannel", namespace),module_ignore_errors=True)["stdout_lines"][0].decode("utf-8")
     return res == '0'
 
-
-def test_po_cleanup(duthosts, rand_one_dut_hostname, tbinfo):
-    """
-    test port channel are cleaned up correctly and teammgrd and teamsyncd process
-    handle  SIGTERM gracefully
-    """
+@pytest.fixture(scope="module", autouse=True)
+def disable_teamd(duthosts, rand_one_dut_hostname, tbinfo):
+    
     duthost = duthosts[rand_one_dut_hostname]
     mg_facts = duthost.get_extended_minigraph_facts(tbinfo)
 
     if len(mg_facts['minigraph_portchannels'].keys()) == 0:
         pytest.skip("Skip test due to there is no portchannel exists in current topology.")
+    yield 
+    # Do config reload to restor everything back
+    logging.info("Reloading config..")
+    config_reload(duthost)
 
-    try:
-        logging.info("Disable Teamd Feature")
-        duthost.shell("sudo systemctl stop teamd")
-        # Check if Linux Kernel Portchannel Interface teamdev are clean up
-        if not wait_until(10, 1, check_kernel_po_interface_cleaned, duthost):
-            fail_msg = "PortChannel interface still exists in kernel"
-            pytest.fail(fail_msg)
-    finally:
-        # Do config reload to restor everything back
-        logging.info("Reloading config..")
-        config_reload(duthost)
+def test_po_cleanup(duthosts, rand_one_dut_hostname, enum_asic_index):
+    """
+    test port channel are cleaned up correctly and teammgrd and teamsyncd process
+    handle  SIGTERM gracefully
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+    logging.info("Disable Teamd Feature")
+    duthost.shell("sudo systemctl stop swss{}".format('@' + str(enum_asic_index) if enum_asic_index is not None else ''))
```

Comment:
get the sonic asic instance and use "stop_service(self, service)" to stop the service.
```
duthost.asic_instance(enum_asic_index).stop_service("swss")